cargo-dist/templates/installer/npm/binary-install.js

```diff
@@ -122,12 +124,27 @@ class Package {
                   );
                 }
               } else if (this.zipExt == ".zip") {
-                const result = spawnSync("unzip", [
-                  "-q",
-                  tempFile,
-                  "-d",
-                  this.installDirectory,
-                ]);
+                let result;
+                if (this.platform.includes("windows")) {
+                  // Windows does not have "unzip" by default on many installations, instead
+                  // we use Expand-Archive from powershell
+                  result = spawnSync("powershell.exe", [
+                    "-NoProfile",
+                    "-NonInteractive",
+                    "-Command",
+                    "& { Expand-Archive -LiteralPath $args[0] -DestinationPath $args[1] -Force }",
```

Comment:
Expand-Archive is safe, but this technique of embedding isn't always so secure.

Essentially, because you're passing down an unquoted string in thru, it could theoretically get expanded into something nasty.  In practice, you'll probably be fine if the inputs are sanitized first (say, by confirming the file exists).

I'd personally feel more comfortable if the command were something like:

~~~PowerShell
& {param([string]$LiteralPath, [string]$DestinationPath) Expand-Archive @psboundParameters } 
~~~

This would force each parameter to be a single string, and pass the parameters directly down to Expand-Path.

Reply:
Great, thank you!